src/umpire/util/Macros.hpp

```diff
@@ -25,13 +25,13 @@
 #include <sstream>
 #include <iostream>
 #include <mutex>
-
-#ifdef UMPIRE_ENABLE_ASSERTS
 #include <cassert>
+
+#ifndef NDEBUG
```

Comment:
I think this bit is redundant, since http://www.cplusplus.com/reference/cassert/assert/ is a no-op when `NDEBUG` is defined, we can just always call it?